src/processor/operator/scan/lookup_node_table.cpp

```diff
@@ -36,5 +36,55 @@ bool LookupNodeTable::getNextTuplesInternal(ExecutionContext* context) {
     return true;
 }
 
+common::idx_t PrimaryKeyScanSharedState::getTableIdx() {
+    std::unique_lock lck{mtx};
+    if (cursor < numTables) {
+        return cursor++;
+    }
+    return numTables;
+}
+
+void PrimaryKeyScanNodeTable::initLocalStateInternal(ResultSet* resultSet,
+    ExecutionContext* context) {
+    ScanTable::initLocalStateInternal(resultSet, context);
+    for (auto& nodeInfo : nodeInfos) {
+        nodeInfo.localScanState = std::make_unique<NodeTableScanState>(nodeInfo.columnIDs);
+        initVectors(*nodeInfo.localScanState, *resultSet);
+    }
+    indexEvaluator->init(*resultSet, context->clientContext->getMemoryManager());
+}
+
+bool PrimaryKeyScanNodeTable::getNextTuplesInternal(ExecutionContext* context) {
+    auto transaction = context->clientContext->getTx();
+    auto tableIdx = sharedState->getTableIdx();
+    if (tableIdx >= nodeInfos.size()) {
+        return false;
+    }
+    KU_ASSERT(tableIdx < nodeInfos.size());
+    auto& nodeInfo = nodeInfos[tableIdx];
+
+    indexEvaluator->evaluate(context->clientContext);
+    auto indexVector = indexEvaluator->resultVector.get();
+    auto& selVector = indexVector->state->getSelVector();
+    KU_ASSERT(selVector.getSelSize() == 1 && selVector.getSelectedPositions()[0] == 0);
```

Comment:
Better directly take `auto pos = selVector.getSelectedPositions()[0]`. The assumption of using `0` as pos here is quite implicit.